### PR TITLE
Support getting external .xml job descriptions

### PR DESCRIPTION
**Pull request by gdhbashton**
This is a new PR to replace #257 whose repo has now been deleted and replaced. The new code should be a little more elegant, and now contains rspec tests.

Cheers,
Gavin.

### CODE REVIEW

File: spec/defines/jenkins_job_spec.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 require 'spec_helper'
 
+Puppet::Util::Log.level = :debug
+Puppet::Util::Log.newdestination(:console)
```
Comment (jhoblitt) [inline spec/defines/jenkins_job_spec.rb]: Why is this needed?

Comment (gdhbashton) [inline spec/defines/jenkins_job_spec.rb]: It's not needed at all. It's debugging code that didn't get removed before I pushed....

File: manifests/job.pp [hunk 2]
```diff
@@ -18,6 +22,7 @@
 #
 define jenkins::job(
   $config,
+  $source = '',
```
Comment (jhoblitt) [inline manifests/job.pp]: Why not make the default `undef`?  I suggest validating the param with `validate_string()`.

File: manifests/job.pp [hunk 3]
```diff
@@ -28,10 +33,18 @@
...
+        config  => $config,
+        jobname => $jobname,
+        enabled => $enabled,
+      }
+    } else {
+      jenkins::job::present { $title:
+        config  => file($source),
+        jobname => $jobname,
+        enabled => $enabled,
+      }
```
Comment (jhoblitt) [inline manifests/job.pp]: Instead of duplicating the resource, I would suggest using a selector to set a variable for the `config` param.
